components/Chat/mermaid/MermaidDiagram.tsx

```diff
@@ -40,15 +44,40 @@ const MermaidDiagramInternal: React.FC<MermaidDiagramProps> = ({ chart, id }) =>
 
   // Render the container pre element (initially hidden, made visible on success)
   return (
-    <pre
-      ref={containerRef}
-      id={uniqueId}
-      className="mermaid"
-      // Hide until library loaded and rendered successfully
-      style={{ visibility: (isLibraryLoaded && !hasError) ? 'visible' : 'hidden', backgroundColor: 'white', border: 'none', padding: '0' }}
-    >
-      {chart}
-    </pre>
+    <div className="relative group">
+      <pre
+        ref={containerRef}
+        id={uniqueId}
+        className="mermaid"
+        // Hide until library loaded and rendered successfully
+        style={{ visibility: (isLibraryLoaded && !hasError) ? 'visible' : 'hidden', backgroundColor: 'white', border: 'none', padding: '0' }}
+      >
+        {chart}
+      </pre>
+      {isLibraryLoaded && !hasError && (
+        <TooltipProvider delayDuration={100}>
+          <Tooltip>
+            <TooltipTrigger asChild>
+              <Button
+                variant="ghost"
+                size="icon"
+                className="absolute top-2 right-2 z-20 md:rounded-xl opacity-0 group-hover:opacity-100 transition-opacity duration-300 hover:bg-gray-200/50"
+                onClick={handleDownload}
+                aria-label="Download image"
+              >
+                <Download className="h-4 w-4 text-gray-700" />
+              </Button>
+            </TooltipTrigger>
+            <TooltipContent>
+              <p>Download</p>
+            </TooltipContent>
+          </Tooltip>
+        </TooltipProvider>
```

Comment:
Is this the same code you just wrote for the image download button?
what would happen if we followed the DRY principle to create a shared <DownloadButton /> component imported by both components?